include/SDL3/SDL_main.h

```diff
@@ -89,6 +89,14 @@
         /* We need to export SDL_main so it can be launched from Java */
         #define SDLMAIN_DECLSPEC    SDL_DECLSPEC
 
+        /* As this is launched from Java, the "real" main it outside of the
+           the binary built from this code. This define makes sure that SDL_main.h
+           and SDL_main_impl.h don't try to provide a native main() function,
+           but if SDL_MAIN_USE_CALLBACKS is used, SDL_main_impl.h is still included
+           for the SDL_main() implementation that handles the callbacks.
+         */
+        #define SDL_MAIN_OUTSIDE_MAIN
```

Comment:
This is very confusing. Let's nuke it and re-add Android below. There will be few enough additional platforms that need this kind of thing that I think we can special case them.

Reply:
on the contrary, we should keep it (though I'm open for a better name) and move the checks for what to do in that case into SDL_main_impl.h

because I just realized that in the SDL_MAIN_USE_CALLBACKS + Android case, currently a standard `int main(int argc, char* argv[])` **will** be generated in SDL_main_impl.h, because
https://github.com/libsdl-org/SDL/blob/b676413657875e1ba3af4eb038611db5deb13d9e/include/SDL3/SDL_main_impl.h#L55
defines `SDL_MAIN_CALLBACK_STANDARD` so after 
https://github.com/libsdl-org/SDL/blob/b676413657875e1ba3af4eb038611db5deb13d9e/include/SDL3/SDL_main_impl.h#L68
we'll run into
https://github.com/libsdl-org/SDL/blob/b676413657875e1ba3af4eb038611db5deb13d9e/include/SDL3/SDL_main_impl.h#L141-L149

So IMO it's best to explicitly define that behavior with its own name in SDL_main.h, and to implement it in SDL_main_impl.h, without having to hardcode the affected platforms in multiple places

(of course either way this needs more changes that I'm currently working on)

Reply:
turned out it's still just one check for android-like platforms, but I still think that having a #define for that behavior in SDL_main.h is both cleaner and potentially useful for users to check than hardcoding SDL_PLATFORM_ANDROID

Reply:
I improved the description to hopefully make it less confusing